### PR TITLE
ENH: Add kernel profiler log for linear regression reduction

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

Adds a kernel profiler log for the whole reduction phase in linear regression.

This revealed that something fishy is going on - it shows the operation taking a lot of time, but the suboperations that already have logs inside it amount to only a small fraction - example after merging https://github.com/uxlfoundation/oneDAL/pull/3435 here:
image

@Alexandr-Solovev Any comments about whether this is working as it should?

---

<details>
<summary>Checklist:</summary>

**Completeness and readability**

- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

</details>

### CODE REVIEW

File: cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i [hunk 1]
```diff
@@ -334,11 +334,14 @@ Status UpdateKernel<algorithmFPType, cpu>::compute(const NumericTable & xTable,
     });
 
     Status st = safeStat.detach();
-    tls.reduce([=, &st](ThreadingTaskType * tlsLocal) -> void {
-        if (!tlsLocal) return;
-        if (st) tlsLocal->reduce(xtx, xty);
-        delete tlsLocal;
-    });
+    {
+        DAAL_PROFILER_TASK(reduction);
```
Comment (Alexandr-Solovev) [inline cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i:338]: Does it make sense to name the task more specific to the algo?

Comment (david-cortes-intel) [reply cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i:338]: Wouldn't it always show under `computeUpdate` either way? Do you have some suggestion for a potential name?

Comment (Alexandr-Solovev) [reply cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i:338]: Yes, it would, then its fine for the name

File: cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i
Comment (Alexandr-Solovev) [inline cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i:316]: I checked locally and looks like this block takes much of the time

Comment (Alexandr-Solovev) [reply cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i:316]: Can you try to also put it wrap it with profiler

Comment (david-cortes-intel) [reply cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i:316]: Before this PR, I tried doing it with direct calls to std::chrono, and the time was spent during the 'reduce' call. That call also has profile logs inside it, and they more or less sum to the total time spent before/after that call.

Comment (Alexandr-Solovev) [reply cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i:316]: on my data reduction 1 is static_threader_for
image

Comment (Alexandr-Solovev) [reply cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i:316]: btw, can you rebase your branch with the latest updates in main

Comment (david-cortes-intel) [reply cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i:316]: Could you try it like this on some server with at least 64 cores and see what you get:
```python
import numpy as np
from time import time
import os
from sklearnex.linear_model import Ridge
import joblib

TRAIN_SIZE=20000
TEST_SIZE=80000
rng = np.random.default_rng(seed=123)
X_train = rng.random(size=(TRAIN_SIZE, 2000))
y_train = rng.random(size=X_train.shape[0])

estimator = Ridge(fit_intercept=True, alpha=2.0)

tm0 = time()
result = estimator.fit(X_train, y_train)
tm1 = time()
print(tm1 - tm0)
```

Comment (david-cortes-intel) [reply cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i:316]: > btw, can you rebase your branch with the latest updates in main

These screenshots were done after merging the PR that fixes kernel profiler bugs btw.

Comment (Alexandr-Solovev) [reply cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i:316]: image
I used the data generated with these parameters
X, y = generate_dataset(
    n_rows_X=20000,
    n_cols_X=2000,
    n_rows_y=20000,
    n_cols_y=1,
    seed=42
)

Comment (david-cortes-intel) [reply cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i:316]: In that screenshot:
* The parts that are under 'reduction_1' sum to more than its parent.
* The parts under 'reduction' are only a small fraction of the parent.

Perhaps it might be more clear if you put calls to std::chrono line-by-line at these points:
https://github.com/uxlfoundation/oneDAL/blob/df4033c3f13d0e20086e424d31fd0148477057fb/cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i#L316
https://github.com/uxlfoundation/oneDAL/blob/df4033c3f13d0e20086e424d31fd0148477057fb/cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i#L336
https://github.com/uxlfoundation/oneDAL/blob/df4033c3f13d0e20086e424d31fd0148477057fb/cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i#L337
https://github.com/uxlfoundation/oneDAL/blob/df4033c3f13d0e20086e424d31fd0148477057fb/cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i#L343
